src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Windows.cs

```diff
@@ -288,6 +289,30 @@ unsafe long GetFileLengthCore()
 
                 if (!Interop.Kernel32.GetFileInformationByHandleEx(this, Interop.Kernel32.FileStandardInfo, &info, (uint)sizeof(Interop.Kernel32.FILE_STANDARD_INFO)))
                 {
+                    if (this._path?.StartsWith(@"\\?\") ?? false)
+                    {
+                        byte[] buffer = ArrayPool<byte>.Shared.Rent(4096);
```

Comment:
Missing return ?

Reply:
Not sure what you mean...

Reply:
You have to call `ArrayPool<byte>.Shared.Return(buffer)` after you use the buffer. And preferably in a `finally`.

Reply:
Updated